### PR TITLE
Number to roman

### PR DESCRIPTION
**Pull request by lloricode**
Number to Roman conversion range of `if ($num < 1 OR $num > 3999)`

output conversion [here](http://webpad.apphb.com/ci4-number-to-roman)

### CODE REVIEW

File: system/Helpers/number_helper.php [hunk 1]
```diff
@@ -242,3 +242,99 @@ function format_number($num, int $precision = 1, string $locale = null, array $o
...
+	* @return string
+	*/
+	function number_to_roman($num)
+	{
+		$num = (int) $num;
+		if ($num < 1 OR $num > 3999)
+		{
+				return;
+		}
+		return _number_to_roman($num, 1);
```
Comment (lonnieezell) [inline system/Helpers/number_helper.php]: I'm not a big fan of having two separate methods here. I see that `_number_to_roman` is called recursively, but as little as the main function does, it seems you could merge the two functions and have them still work. Please try to do that then we'll consider it.

File: system/Helpers/inflector_helper.php [hunk 2]
```diff
@@ -93,7 +93,7 @@ function singular(string $string): string
 					'/(s)tatuses$/'														 => '\1\2tatus',
 					'/(c)hildren$/'														 => '\1\2hild',
 					'/(n)ews$/'															 => '\1\2ews',
-					'/([^us])s$/'														 => '\1'
+					'/([^us])s$/'														 => '\1'
```
Comment (lonnieezell) [inline system/Helpers/inflector_helper.php]: You've got a couple of places here with whitespace at the end of a line. Our styleguide frowns on that. Please remove.

**Comment by lloricode** [comment]
ok sir, I will fix this when I open my laptop.

**Comment by lloricode** [comment]
I can't run this `./phpunit tests/system/Helpers/NumberHelperTest`.

screenshot at 2017-07-25 14-49-23

**Comment by lonnieezell** [comment]
It looks like you can't run those tests because you don't have the `intl` extension installed in your version of PHP. That's becoming a requirement for CI4. 

You can always run just a single test, though: 

```
./vendor/bin/phpunit tests/system/Helpers/NumberHelperTest.php --filter=test_roman_number
```

**Comment by lloricode** [comment]
ok sir., I will do this, thanks. in info.. I will resubmit the propos later.

**Comment by lloricode** [comment]
I can explain fluent but maybe they understand what it mean

**Comment by jim-parry** [comment]
not to be too picky, but we expect commits to be GPG-signed :)

**Comment by lloricode** [comment]
@jim-parry im not familiar with the term `GPG-signed` .

**Comment by lonnieezell** [comment]
Here's what he's talking about: 

https://bcit-ci.github.io/CodeIgniter4/contributing/signing.html

**Comment by lonnieezell** [comment]
That doc works for now. thanks! We can always expand on it later if need be.

**Comment by lloricode** [comment]
Thanks for new idea @lonnieezell.

**Pull request merged**